py/selenium/webdriver/common/bidi/network.py

```diff
@@ -0,0 +1,326 @@
+# Licensed to the Software Freedom Conservancy (SFC) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The SFC licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+import uuid
+
+from .session import session_subscribe
+from .session import session_unsubscribe
+
+
+class Network:
+    EVENTS = {
+        "before_request": "network.beforeRequestSent",
+        "response_started": "network.responseStarted",
+        "response_completed": "network.responseCompleted",
+        "auth_required": "network.authRequired",
+        "fetch_error": "network.fetchError",
+    }
+
+    PHASES = {
+        "before_request": "beforeRequestSent",
+        "response_started": "responseStarted",
+        "auth_required": "authRequired",
+    }
+
+    def __init__(self, conn):
+        self.conn = conn
+        self.callbacks = {}
+        self.subscriptions = {
+            "network.responseStarted": [],
+            "network.beforeRequestSent": [],
+            "network.authRequired": [],
+        }
+
+    def _command_iterator(self, command):
+        """Generator to yield command."""
+        yield command
+
+    def has_callbacks(self):
+        """Checks if there are any callbacks set."""
+        return len(self.callbacks) > 0
+
+    def __add_intercept(self, phases=None, contexts=None, url_patterns=None):
+        """Add an intercept to the network.
+
+        Parameters:
+        ----------
+            phases (list, optional): A list of phases to intercept.
+                Default is None.
+            contexts (list, optional): A list of contexts to intercept.
+                Default is None.
+            url_patterns (list, optional): A list of URL patterns to intercept.
+                Default is None.
+        """
+        if phases is None:
+            phases = []
+        if contexts is None and url_patterns is None:
+            params = {
+                "phases": phases,
+            }
+        elif contexts is None:
+            params = {"phases": phases, "urlPatterns": url_patterns}
+        elif url_patterns is None:
+            params = {"phases": phases, "contexts": contexts}
+        else:
+            params = {"phases": phases, "contexts": contexts, "urlPatterns": url_patterns}
+        command = {"method": "network.addIntercept", "params": params}
```

Comment:
We should either use `Command.ADD_INTERCEPT` here in the method since we are declaring it in `command.py` (which is currently not utilized) or remove those commands from `command.py`.